org.opentosca.container.api/src/org/opentosca/container/api/service/InstanceService.java

```diff
@@ -726,6 +730,31 @@ public SituationTriggerInstance getSituationTriggerInstance(final Long id) {
         throw new RuntimeException("SituationTriggerInstance <" + id + "> not found.");
     }
 
+    public SituationsMonitor createNewSituationsMonitor(final ServiceTemplateInstance instance,
+                                                        final Map<String,Collection<Long>> situations) {
+        final SituationsMonitor monitor = new SituationsMonitor();
+
+        monitor.setServiceInstance(instance);                
+      
+        monitor.setNode2Situations(situations);        
+
+        this.situationsMonitorRepo.add(monitor);
+        return monitor;
+    }
+
+    public Collection<SituationsMonitor> getSituationsMonitors() {
+        return this.situationsMonitorRepo.findAll();
+    }
+
+    public Collection<SituationsMonitor> getSituationsMonitors(final Long serviceInstanceId) {
```

Comment:
`return this.getSituationsMonitors().stream().filter(monitor -> monitor.getServiceInstance() != null && monitor.getServiceInstance().getId().equals(serviceInstanceId)).collect(Collectors.toList());` ?

But I guess you don´t like that :-)

Reply:
done